common/ripple/userUtils.py

```diff
@@ -753,46 +553,49 @@ async def compareHWID(userID: int, mac: str, unique: str, disk: str) -> bool:
     )
 
 
-async def logHardware(userID: int, hashes: list[str], activation: bool = False) -> bool:
+def validate_hwid_set(hwid_set: list[str]) -> bool:
+    """Validate that the mac addresses, unique id, and disk id are present."""
+    return all(hwid_set[2:5])
+
+
+async def associate_user_with_hwids_and_restrict_if_multiaccounting(
+    user_id: int,
+    # TODO: refactor hwid sets into an object across the codebase
+    hwid_set: list[str],
+    *,
+    associate_with_account_activation: bool = False,
+) -> None:
     """
-    Hardware log
-    USED FOR MULTIACCOUNT DETECTION
-
-    :param userID: user id
-    :param hashes:	Peppy's botnet (client data) structure (new line = "|", already split)
-                    [0] osu! version
-                    [1] plain mac addressed, separated by "."
-                    [2] mac addresses hash set
-                    [3] unique ID
-                    [4] disk ID
-    :param activation: if True, set this hash as used for activation. Default: False.
-    :return: True if hw is not banned, otherwise false
+    Associate a user with a given set of hardware identifiers.
+
+    This function *may* restrict users who are determined to be engaging
+    in multi-accounting.
+
+    If `associate_with_account_activation` is True, set this hash as 'used for activation'.
+
+    The hashset comes in the following form:
+      - [0]: osu! version
+      - [1]: plain mac addressed, separated by "."
+      - [2]: mac addresses hash set
+      - [3]: unique ID
+      - [4]: disk ID
     """
 
     # Get username
-    username = await getUsername(userID)
-
-    # Make sure the strings are not empty
-    for i in hashes[2:5]:
-        if not i:
-            await audit_logs.send_log_as_discord_webhook(
-                message=f"Invalid hash set ({hashes}) for user [{username}](https://akatsuki.gg/u/{userID}) in HWID check",
-                discord_channel="ac_confidential",
-            )
-            return False
+    username = await get_username_from_id(user_id)
 
     # Run some HWID checks on that user if he is not restricted
-    if not await isRestricted(userID):
+    if not await is_restricted(user_id):
         """
-        compare_ids = compareHWID(userID, hashes[2], hashes[3], hashes[4])
+        compare_ids = compare_hwid(user_id, hwid_set[2], hwid_set[3], hwid_set[4])
 
         if not compare_ids: # Remove cmyui permissions if on a HWID different than usual.. Just safety procautions..
-            log.anticheat("{}: Unusual login detected.\n\nHashes:\nMAC: {}\nUnique: {}\nDisk: {}\n\nTheir login has been disallowed.".format(userID, hashes[2], hashes[3], hashes[4]), 'ac_confidential')
+            log.anticheat("{}: Unusual login detected.\n\nHashes:\nMAC: {}\nUnique: {}\nDisk: {}\n\nTheir login has been disallowed.".format(user_id, hwid_set[2], hwid_set[3], hwid_set[4]), 'ac_confidential')
             return False

```

Comment:
Interesting that these checks are disabled -- I wonder when this came to be? I assume ripple is running these checks